service/src/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java

```diff
@@ -96,6 +108,29 @@ public void run() {
       String msg = "Starting " + ThriftBinaryCLIService.class.getSimpleName() + " on port "
           + portNum + " with " + minWorkerThreads + "..." + maxWorkerThreads + " worker threads";
       LOG.info(msg);
+
+      // New thread : Custom authentication with Kerberos thread
+      final ThriftCLIService svc = this;
```

Comment:
what is the point of this declaration?
can you more simply reference 'this' from startCustomWithKerberos() below and drop the param altogether?